script.js

```diff
@@ -1,4 +1,190 @@
+  }
+  // Player 2 wins
+  if (allPlayerScores[0] < allPlayerScores[1]) {
+    compareMessage =
+      compareMessage +
+      " <br>Player 2 wins! <br>Please press the Submit button to reset the game.<br>";
+  }
+  // Draw
+  if (allPlayerScores[0] == allPlayerScores[1]) {
+    compareMessage =
+      compareMessage +
+      " <br>It is a Draw! <br>Please press the Submit button to reset the game.<br>";
+  }
+  return compareMessage;
+};
+
+// Helper Function 5 - Reset Game
+var resetGame = function () {
+  var resetGameMessage = "";
+  currentPlayer = 1;
+  gameState = gameStateDiceRoll;
+  allPlayerScores = [];
+};
+
+// Helper Function 6 - Player Score Summation and Record
+var player1Sum = function () {
+  player1ScoreRecord.push(allPlayerScores[0]);
+  player1ScoreSum += allPlayerScores[0];
+  return player1ScoreSum;
+};
+var player2Sum = function () {
+  player2ScoreRecord.push(allPlayerScores[1]);
+  player2ScoreSum += allPlayerScores[1];
+  return player2ScoreSum;
+};
+
+//Helper Function 7 - Player Score Sum Comparison
+var comparePlayerScoreSums = function () {
+  var compareSumMessage = "";
+
+  // Score Sum Draw
+  if (player1ScoreSum == player2ScoreSum) {
+    compareSumMessage = `<br>It is a draw thus far.`;
+  }
+
+  // Score Sum - Player 1 win
+  if (player1ScoreSum > player2ScoreSum) {
+    compareSumMessage = `<br>Player 1 is currently winning!`;
+  }
+
+  // Score Sum - Player 2 win
+  if (player1ScoreSum < player2ScoreSum) {
+    compareSumMessage = `<br>Player 2 is currently winning!`;
+  }
+
+  return compareSumMessage;
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  var gameMessage = " ";
+  // Roll Dice
+  if (gameState == gameStateDiceRoll) {
+    gameMessage = rollDiceForPlayer();
+    gameState = gameStateChooseDiceOrder;
+  }
+  // Player Score
+  else if (gameState == gameStateChooseDiceOrder) {
+    gameMessage = getPlayerScore(input);
+    if (currentPlayer == 1 && inputValidation == validInput) {
+      // Change Player
+      currentPlayer = 2;
+      gameState = gameStateDiceRoll;
+      return (
+        gameMessage + " It is Player 2's turn. Please switch over the controls."
+      );
+    }
+    // Shift to Compare Score Mode
+    if (currentPlayer == 2 && inputValidation == validInput) {
+      gameState = gameStateCompareScores;
+      return gameMessage + " Press the Submit button to compare scores.";
+    }
+  }
+  // Compare Scores
+  else if (gameState == gameStateCompareScores) {
+    gameMessage =
+      comparePlayerScores() +
+      `<br>SCOREBOARD<br>` +
+      `<br>Player 1's combined score thus far is ${player1Sum()}. <br>Player 2's combined score thus far is ${player2Sum()}.` +
+      comparePlayerScoreSums() +
+      `<br><br>Number of rounds played = ${gamesPlayed}. <br>Player 1's 2-digit numbers thus far are ${player1ScoreRecord}. <br>Player 2's 2-digit numbers thus far are ${player2ScoreRecord}.`;
+    gamesPlayed += 1;
```

Comment:
Great job structuring your main function to show the flow of the game and keeping all logic in the helper functions!